src/Project.js

```diff
@@ -547,20 +547,56 @@ class Project extends DirItem {
         return 100 / (1.0 + demeritPoints/base);
     }
 
+    /**
+     * Sort the results:
+     *
+     * 1. alphabetically by source file path
+     * 2. numerically by line number within the source file
+     *
+     * The results array will be sorted in place.
+     *
+     * @param {Array.<Result>} results the results array to sort
+     */
+    sortResults(results) {
```

Comment:
Placement and signature of this method does not really make sense to me. It's added as a public void method of `Project` instance, which to me would suggest that it somehow relies on internal state of the project or modifies it. Furthermore, it accepts an external `Result[]` (i.e. provided by the caller), but it does not really impact the project because this method just semi-silently (until you read the comment ofc) mutates the input it's been given. To me, this is counterintuitive.

I would rather have a separate `ResultsComparator`
```js
export const ResultsComparator = (/** @type {Result} */ left, /** @type {Result} */ right) => {
    /* ... */
    return 0;
}
export default ResultsComparator;
```
and use that like
```js
const results = this.findIssues();
results.sort(ResultsComparator);
```
(or even `toSorted()` if not for the fact it's available only since Node 20)

this way, _how_ results are sorted is not a concern of `Project` anymore. Also comparator could (later?) become a property of the output formatter, because I'm not entirely sure `Project` has any reason to know this stuff directly.

Wdyt?

Reply:
done